WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -0,0 +1,191 @@
+using Microsoft.Extensions.Hosting;
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Crypto.Randomness;
+using WalletWasabi.Tor.Http;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+using WalletWasabi.WabiSabi.Crypto;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class CoinJoinClient : BackgroundService, IDisposable
+	{
+		private bool _disposedValue;
+		private CredentialPool AmountCredentialPool { get; } = new();
+		private CredentialPool VsizeCredentialPool { get; } = new();
+		private Round Round { get; }
```

Comment:
This is not wrong, we can have a `RoundState` (something shared by the server and client) or, we can have a `ClientRound` as we have in v1.

Reply:
Oops, i missed that... we definitely need a different class, although they can share a base class or interface. The existing `Round` contains server only concepts. The `MultipartyTransaction` model stuff is designed to be shared, but it's only a part of what is needed.

Reply:
So another Round class under namespace WalletWasabi.WabiSabi.Client?